### PR TITLE
feat: Add example Go stack for Step Functions

### PR DESCRIPTION
**Pull request by lym953**
### What does this PR do?
Add an example Go stack that creates and instruments a State Machine, which invokes a child State Machine and a Lambda function, with trace merging enabled for both.

### Testing Guidelines
Deployed the stack, and the step function showed up in Datadog app, including its traces.
DatadogCdkGoStack

### Types of Changes

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [x] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog

### CODE REVIEW

File: examples/go-stack/app.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ func NewAppStackWithoutDatadog(scope constructs.Construct, id *string, props *Ap
 	stack := awscdk.NewStack(scope, id, &sprops)
 
 	myFunction := awslambda.NewFunction(stack, jsii.String("HelloWorldFunction"), &awslambda.FunctionProps{
-		Runtime: awslambda.Runtime_NODEJS_20_X(), // Provide any supported Node.js runtime
```
Comment (lym953) [inline examples/go-stack/app.go:26]: I think this comment was copy-pasted from a template. Redundant here.

File: examples/go-stack/app.go [hunk 2]
```diff
@@ -132,7 +132,7 @@ func main() {
 func env() *awscdk.Environment {
 	env := awscdk.Environment{
 		Account: jsii.String("425362996713"),
-		Region:  jsii.String("us-east-1"),
+		Region:  jsii.String("sa-east-1"),
```
Comment (lym953) [inline examples/go-stack/app.go:135]: We use this region the most.

File: examples/go-stack/go.mod [hunk 3]
```diff
@@ -3,7 +3,7 @@ module app
 go 1.18
 
 require (
-	github.com/DataDog/datadog-cdk-constructs-go/ddcdkconstruct v1.13.0
+	github.com/DataDog/datadog-cdk-constructs-go/ddcdkconstruct v1.18.0
```
Comment (lym953) [inline examples/go-stack/go.mod:6]: Make the existing example stack use the latest version of DD CDK Construct.

File: examples/go-stack/go.sum [hunk 4]
```diff
@@ -1,5 +1,7 @@
-github.com/DataDog/datadog-cdk-constructs-go/ddcdkconstruct v1.13.0 h1:AOgs4MEGLb0gmkEdBx2HC7BrUtrpM46tyET+cJ7t80M=
+github.com/DataDog/datadog-cdk-constructs-go/ddcdkconstruct v1.13.0 h1:K4HIqyzkJSDltkTk5A9eqsVCrtHI0u4XxkKl7TB0nAM=
```
Comment (lym953) [inline examples/go-stack/go.sum:1]: Changes made by `go get`

File: examples/step-functions-go-stack/.gitignore [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+# go.sum should be committed
+!go.sum
+
+# CDK asset staging directory
+.cdk.staging
+cdk.out
+
+# The copied Go package for dev testing
+ddcdkconstruct/
```
Comment (lym953) [inline examples/step-functions-go-stack/.gitignore:23]: This is added by me. Other lines are from AWS Go project template.

**Review by cswatt — APPROVED** [review]

**Review by avedmala — APPROVED** [review]
Great example, thanks Yiming!

**Pull request merged**